### ISSUE TITLE
Force closing KSP while sending an important message (such as scenario ones) crashes KSP

### ISSUE THREAD

**Issue by RockyTV** [godarklight/DarkMultiPlayer#317]
Last lines of my output_log.txt:

```
(Filename: C:/buildslave/unity/build/artifacts/StandalonePlayerGenerated/UnityEngineDebug.cpp Line: 56)

[2251.934] DarkMultiPlayer: Sending 1 scenario modules
 
(Filename: C:/buildslave/unity/build/artifacts/StandalonePlayerGenerated/UnityEngineDebug.cpp Line: 56)

NullReferenceException
  at (wrapper managed-to-native) UnityEngine.Component:InternalGetTransform ()

  at UnityEngine.Component.get_transform () [0x00000] in <filename unknown>:0 

  at PQSCity.OnSphereReset () [0x00000] in <filename unknown>:0 

  at PQS.Mod_OnSphereReset () [0x00000] in <filename unknown>:0 

  at PQS.ResetSphere () [0x00000] in <filename unknown>:0 

  at PQS.OnDestroy () [0x00000] in <filename unknown>:0 
 
(Filename:  Line: -1)
```

Stack trace:

```
========== OUTPUTING STACK TRACE ==================

(0x0139EF68) (KSP): (filename not available): Behaviour::Transfer<SafeBinaryRead> + 0x734e8
(0x040EC849) (Mono JIT code): (filename not available):  (wrapper managed-to-native) UnityEngine.Time:get_realtimeSinceStartup () + 0x21 (040EC828 040EC876) [00F44E70 - Unity Root Domain] + 0x0
(0x08A4A719) (Mono JIT code): (filename not available):  DarkMultiPlayer.NetworkWorker:CheckDisconnection () + 0x31 (08A4A6E8 08A4A759) [00F44E70 - Unity Root Domain] + 0x0
(0x08A4A601) (Mono JIT code): (filename not available):  DarkMultiPlayer.NetworkWorker:SendThreadMain () + 0x21 (08A4A5E0 08A4A6AD) [00F44E70 - Unity Root Domain] + 0x0
(0x0408E571) (Mono JIT code): (filename not available):  (wrapper runtime-invoke) object:runtime_invoke_void__this__ (object,intptr,intptr,intptr) + 0x41 (0408E530 0408E5C5) [00F44E70 - Unity Root Domain] + 0x0
(0x100F0542) (mono): (filename not available): mono_set_defaults + 0x22cb
(0x1005D7F4) (mono): (filename not available): mono_runtime_invoke + 0x51
(0x1005DC10) (mono): (filename not available): mono_runtime_delegate_invoke + 0x44
(0x1007DB0E) (mono): (filename not available): mono_thread_interruption_request_flag + 0x440
(0x1010B66C) (mono): (filename not available): mono_gc_max_generation + 0x2646
(0x76DCA534) (KERNEL32): (filename not available): BaseThreadInitThunk + 0xe
(0x77D68F8B) (ntdll): (filename not available): RtlInitializeExceptionChain + 0x84
(0x77D68F61) (ntdll): (filename not available): RtlInitializeExceptionChain + 0x5a

========== END OF STACKTRACE ===========
```

**Comment by oldmud0** [comment]
Solution: disconnect first

**Comment by RockyTV** [comment]
Fixed via 6192598
Closed #317.
